lib/p2p/Pool.ts

```diff
@@ -382,7 +382,12 @@ class Pool extends EventEmitter {
    * @param nodePubKey The node pub key of the node for which to get reputation information
    * @return true if the specified node exists and the event was added, false otherwise
    */
-  public getNodeReputation = async (nodePubKey: string): Promise<NodeReputationInfo> => {
+  public getNodeReputation = async (nodeIdentifier: string): Promise<NodeReputationInfo> => {
```

Comment:
I think that ideally the method signatures for this and other methods in the Pool class that accept `nodePubKey` should not change. For one thing there are quite a few places where the parameter comments and naming fall out of sync. But mainly I don't think these methods should be concerned with aliases, which is there for display purposes and user convenience, but not internal logic.

Rather, the logic to determine whether the `node_identifier` we were provided in an rpc request is an alias or a pub key should exist outside of these methods and be used before we call them.

Maybe there can be an `aliasUtils.ts` file (which could also house the `getAlias` method that's currently with uri utils) that has a method which determines whether a node identifier is an alias (by looking at length as you're currently doing). You can call that method from Service.ts when dealing with a nodeIdentifier, then if it turns out it's an alias we make another call to resolve the alias like so:

```typescript
const nodePubKey = isAlias(nodeIdentifier) ? this.pool.resolveAlias(nodeIdentifier) : nodeIdentifier;
```

We'd also probably need to throw an error when the alias resolves to `undefined` and add that error to the list of codes corresponding to `INVALID_ARGUMENT` in getGrpcError.ts, so that when people make an rpc call with an invalid or unknown alias, they get an error message accurately indicating what was wrong with the call.